x-pack/plugins/index_management/public/application/services/routing.ts

```diff
@@ -4,35 +4,31 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { BASE_PATH } from '../../../common/constants';
-import { IndexTemplateFormatVersion } from '../../../common';
 
 export const getTemplateListLink = () => {
   return `${BASE_PATH}templates`;
 };
 
 // Need to add some additonal encoding/decoding logic to work with React Router
 // For background, see: https://github.com/ReactTraining/history/issues/505
-export const getTemplateDetailsLink = (
-  name: string,
-  formatVersion: IndexTemplateFormatVersion,
-  withHash = false
-) => {
-  const baseUrl = `${BASE_PATH}templates/${encodeURIComponent(
-    encodeURIComponent(name)
-  )}?v=${formatVersion}`;
-  const url = withHash ? `#${baseUrl}` : baseUrl;
+export const getTemplateDetailsLink = (name: string, isLegacy?: boolean, withHash = false) => {
+  const baseUrl = `${BASE_PATH}templates/${encodeURIComponent(encodeURIComponent(name))}`;
+  let url = withHash ? `#${baseUrl}` : baseUrl;
+  if (isLegacy) {
+    url = `${url}?legacy=${isLegacy}`;
+  }
   return encodeURI(url);
 };
 
-export const getTemplateEditLink = (name: string, formatVersion: IndexTemplateFormatVersion) => {
+export const getTemplateEditLink = (name: string, isLegacy?: boolean) => {
   return encodeURI(
-    `${BASE_PATH}edit_template/${encodeURIComponent(encodeURIComponent(name))}?v=${formatVersion}`
+    `${BASE_PATH}edit_template/${encodeURIComponent(encodeURIComponent(name))}?legacy=${isLegacy}`
```

Comment:
Since legacy is marked as optional, should this be something like:

```
isLegacy ? `?legacy=${isLegacy}` : ''
```

Reply:
It would be a cleaner URL yes, but the logic would be the same as we would get 

```
?legacy=undefined
```

So checking for 

```js
const isLegacy = queryParam.legacy === 'true'; 
```

would give false.

Reply:
I made the change to this:

```js
?legacy=${isLegacy === true}`
```